osquery/core/conversions.h

```diff
@@ -156,6 +156,20 @@ inline Status safeStrtol(const std::string& rep, size_t base, long int& out) {
   return Status(0);
 }
 
+/// Safely convert a std::wstring to an integer
+inline int safeWstrToInt(std::wstring str) {
+  //std::stoi can throw, and std::stol doesn't support std::wstring
+  try {
+    return std::stoi(str);
+  }
+  catch (const std::out_of_range&) {
+    return 0;
+  }
+  catch (const std::invalid_argument&) {
+    return 0;
+  }
+}
+
```

Comment:
There is a new way to convert string to numbers in `conversions.h`: `tryTo<in>(...)`. Check it out :)

Reply:
@akindyakov sure thing!